src/ai/defaultai.cc

```diff
@@ -3999,9 +3999,7 @@ bool DefaultAI::dispensable_road_test(const Widelands::Road& road) {
 		}
 	}
 
-	if (spots_ > kSpotsEnough && wares_on_road > 5) {
-		return false;
-	} else if (wares_on_road > 8) {
+	if (wares_on_road > (spots_ > kSpotsEnough ? 5 : 8)) {
```

Comment:
This is hard to read. How about this? (or will it trigger a clang-tidy warning again?)
```suggestion
	if (wares_on_road > 8 || (spots_ > kSpotsEnough && wares_on_road > 5)) {
```